packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp

```diff
@@ -679,16 +813,37 @@ static EdgeInsets calculateOverflowInset(
   return overflowInset;
 }
 
+void YogaLayoutableShadowNode::setContentsLayoutMetrics(LayoutContext layoutContext) {
+  if (layoutMetrics_ != EmptyLayoutMetrics) {
+    setLayoutMetrics(EmptyLayoutMetrics);
+  }
+
+  for (const auto &child : getChildren()) {
+    if (const auto& layoutableChild = std::dynamic_pointer_cast<const YogaLayoutableShadowNode>(child)) {
+      if (layoutableChild->hasDisplayContentsStyle()) {
+        auto& contentsNode = shadowNodeFromContext(&layoutableChild->yogaNode_);
+        contentsNode.setContentsLayoutMetrics(layoutContext);
+      }
+    }
+  }
```

Comment:
Why do we need to do this if we still iterate through children in layout()?

Reply:
Layout info is only propagated for nodes that are present in the Yoga tree. Since nodes with `display: contents` are not present there, they were effectively skipped.